interfaces/builtin/posix_mq.go

```diff
@@ -0,0 +1,237 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const posixMQSummary = `allows access to POSIX message queues`
+
+const posixMQBaseDeclarationSlots = `
+  posix-mq:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: false
+`
+
+const posixMQPermanentSlotSecComp = `
+mq_open
+mq_getsetattr
+mq_unlink
+mq_notify
+mq_timedreceive
+mq_timedsend
+`
+
+var posixMQPlugPermissions = []string{
+	"open",
+	"associate", /* alias for open */
+	"read",
+	"receive", /* alias for read */
+	"write",
+	"send", /* alias for write */
+	"create",
+	"delete",
+	"destroy", /* alias for delete */
```

Comment:
why do we need the aliases?

Reply:
I'm not sure - they were part of the prerelease AppArmor spec. I can remove them to clean it up a bit, if it's preferred to have fewer

Reply:
Let's wait and see what @pedronis says :-)

Reply:
yes, unless we break something that shipped, we definitely don't want to support evolutionary leftovers

Reply:
ACK - will remove the aliases